modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java

```diff
@@ -671,6 +681,27 @@ public ClientProcessorMXBean mxBean() {
         return new ClientProcessorMXBeanImpl();
     }
 
+    /** */
+    private void onOutboundMessageOffered(GridNioSession ses, int queueSize) {
+        if (queueSize < cliConnCfg.getSessionOutboundMessageQueueLimit())
+            return;
+
+        srv.close(ses).listen(fut -> {
+            if (fut.error() == null && fut.result()) {
+                U.quietAndWarn(log, "Ignite Thin Client outbound message queue size is exceeded" +
+                    " 'SessionOutboundMessageQueueLimit', it will be disconnected" +
+                    " [locNodeId=" + ctx.localNodeId() +
+                    ", clientAddress=" + ses.remoteAddress() +
+                    ", sessionOutboundMessageQueueLimit=" + cliConnCfg.getSessionOutboundMessageQueueLimit() + ']');
+            }
+        });
+    }
+
+    /** */
+    GridNioServer<ClientMessage> nioServer() {
```

Comment:
I suppose that introducing a getter just for test purposes is not needed. You can get this field by reflection.